src/Optical/Emitters/Emitters.jl

```diff
@@ -0,0 +1,41 @@
+#=
+Emitters are defined by Pixels and SpatialLayouts. An emitter has a spectrum, and an optical power distribution over the hemisphere. 
+These are intrinsic physical properties of the emitter.
+=#
+
+module Emitters
+
+using ...OpticSim, ..Geometry
+using LinearAlgebra
+
+struct MissingImplementationException <: Exception
+    msg::String
+end
+
+# defining name placeholders to override in nested modules
```

Comment:
Now I understand why these functions are defined here because otherwise when they are overriden in other modules you will get a compiler error saying the function is not defined.

This seems like a **BAD** idea. Not standard Julia style, hard to understand, scatters the function definitions for a module across other unrelated modules. If there are multiple definitions for the same function in multiple modules (let's hope this never happens) the resulting function definition will be the last one the compiler executes. This is difficult for a developer to predict since it requires tracking in what order the compiler compiles the modules.

These modules should be re-architected to eliminate the need to do this.